Sources/Sentry/SentryGeo.m

```diff
@@ -0,0 +1,104 @@
+#import "SentryGeo.h"
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@implementation SentryGeo
+
+- (instancetype)init
+{
+    return [super init];
+}
+
+- (id)copyWithZone:(nullable NSZone *)zone
+{
+    SentryGeo *copy = [[SentryGeo allocWithZone:zone] init];
+
+    @synchronized(self) {
+        if (copy != nil) {
+            copy.city = self.city;
+            copy.countryCode = self.countryCode;
+            copy.region = self.region;
+        }
+    }
+
+    return copy;
+}
+
+- (NSDictionary<NSString *, id> *)serialize
+{
+    NSMutableDictionary *serializedData = [NSMutableDictionary dictionary];
+
+    @synchronized(self) {
+        serializedData[@"city"] = self.city;
+        serializedData[@"country_code"] = self.countryCode;
+        serializedData[@"region"] = self.region;
+    }
+
+    return serializedData;
+}
+
+- (BOOL)isEqual:(id _Nullable)other
+{
+    @synchronized(self) {
+        if (other == self) {
+            return YES;
+        }
+        if (!other || ![[other class] isEqual:[self class]]) {
+            return NO;
+        }
+
+        return [self isEqualToGeo:other];
+    }
+}
+
+- (BOOL)isEqualToGeo:(SentryGeo *)geo
+{
+    @synchronized(self) {
+        // We need to get some local copies of the properties, because they could be modified during
+        // the if statements
```

Comment:
I know this comment is copypasted from SentryUser, but I'm not sure what it describes is sufficient, since those properties could just as easily change in between those copy operations as they could during evaluation of the `if` statements:
1. Call `[a isEqualToGeo:b]`
2. copy `b.city` into `otherCity`
3. something modifies `b.countryCode` while comparing `a.city` to `otherCity`
4. copy `b.countryCode` into `otherCountryCode`

Now we're in an inconsistent state because `b` changed while it was being compared to `a`, and maybe has a city of London, England, but now has a country code of Japan. Is this a problem @philipphofmann?

I wonder if it doesn't make more sense to just compare their hashes after the self/nil checks, to implicitly compare their member data, since those are created in `@synchronized` blocks.

Reply:
Is this an issue that is occurring with the other classes that follow this approach? If so we should probably look at this throughout the SDK, which would warrant a seperate PR. If there's no issue i'd rather leave the approach the same. WDYT?

Reply:
Yes, I agree it would be best in a separate issue/PR. I'll split out further discussion elsewhere.

Reply:
I think we can remove the comment. I can't recall why I added it with https://github.com/getsentry/sentry-cocoa/pull/888. I think what you described in https://github.com/getsentry/sentry-cocoa/pull/2710#discussion_r1127523845 shouldn't be a problem. How likely is it that somebody modifies the user from multiple threads simultaneously and calls equals on them? I think we can remove all the `@synchronized` blocks in the user, but we should do this in another PR. @denrase, I think you should remove the `synchronized` for the SentryGeo class before merging this PR.

Reply:
I removed the synchronized from the user https://github.com/getsentry/sentry-cocoa/pull/2809.